### ISSUE TITLE
Reading `particlePatches["numParticles"]` fails

### ISSUE THREAD

**Issue by psychocoderHPC** [openPMD/openPMD-api#907]
**Describe the bug**
I am testing https://github.com/openPMD/openPMD-api/tree/9e201df4e20f3cf41b75df55a19ae9a33f57cdae together with PIConGPU und realized an error when loading an old checkpoint.
After investing some hours in debugging the issue I tested the release `0.12.0-alpha` and all works fine.

The error is:
```
Unhandled exception of type 'St13runtime_error' with message 'Chunk does not reside inside dataset (Dimension on index 0. DS: 262144 - Chunk: 28299152)', terminating
```

**To Reproduce**
You can use this dataset [checkpoint_blosc_000100.bp.zip](https://github.com/openPMD/openPMD-api/files/5857933/checkpoint_blosc_000100.bp.zip) (the dataset is not compressed, please ignore that it is named blosc)
The dataset was produced with a single MPI rank.

The data is read with the C++ API: https://github.com/ComputationalRadiationPhysics/picongpu/blob/b17941d2b90afb83b17284cae1b70e62d33e088e/include/picongpu/plugins/openPMD/restart/LoadSpecies.hpp#L108-L110

ADIOS2 internal tools shows that the data in the file is correct:
```
bpls checkpoint_blosc_000100.bp --dump -a /data/100/particles/e/particlePatches/numParticles
  uint64_t  /data/100/particles/e/particlePatches/numParticles               {1}
    (0)    262144 
```

**Expected behavior**

**Software Environment**
 - version of openPMD-api: https://github.com/openPMD/openPMD-api/tree/9e201df4e20f3cf41b75df55a19ae9a33f57cdae (current openpmd-api dev branch)
 - installed openPMD-api via: self compiled
 - operating system: Red Hat Enterprise Linux release 8.2 (Ootpa)
 - machine: closed system
 - name and version of Python implementation: [e.g. CPython 3.8]
 - version of ADIOS2: https://github.com/ornladios/ADIOS2/tree/dd822e30103f04418637688bd5cda362be28b0c4
 - name and version of MPI: [e.g. OpenMPI 4.1.0]

**Additional context**
As I wrote above after switching from the current dev branch to `0.12.0-alpha` all works fine.
For unknown reason openpmd-api is reading `28299152` from the dataset instead of `262144` what is stored in the adios file.

**Comment by psychocoderHPC** [comment]
In PIConGPU we implemented that we can support openpmd-api `0.12.0-alpha` and `0.13.x` in this [PR](https://github.com/ComputationalRadiationPhysics/picongpu/pull/3485/)
On some places like https://github.com/ComputationalRadiationPhysics/picongpu/blob/b17941d2b90afb83b17284cae1b70e62d33e088e/include/picongpu/plugins/openPMD/restart/LoadSpecies.hpp#L109 we have not applied our macro `WRITE_ITERATIONS`. 
I am not sure if there is a need for it. Never the less I tested using `WRITE_ITERATIONS` on all missed places where we read data but it was not solving the issue.

@franzpoeschel could you have a look. The macro is called `WRITE_ITERATIONS`, do we need to use it also in places where we read data with openpmd-api?

**Comment by psychocoderHPC** [comment]
update: I tested now the last stable release `0.13.1` and have the reading error from the dev branch too.

**Comment by franzpoeschel** [comment]
Fixed with #909, thanks for reporting!

Also, `WRITE_ITERATIONS` is a macro defined in PIConGPU, not in openPMD. We use it to support writing iterations in 0.12.* as well as in 0.13.* streaming mode. It has nothing to do with this issue.
